apps/ec.c

```diff
@@ -219,6 +219,11 @@ int ec_main(int argc, char **argv)
 
     if (no_public)
         EC_KEY_set_enc_flags(eckey, EC_PKEY_NO_PUBKEY);
+    else {
+        unsigned int enc_flags = EC_KEY_get_enc_flags(eckey);
+        enc_flags &= ~EC_PKEY_NO_PUBKEY,
+        EC_KEY_set_enc_flags(eckey, enc_flags);
+    }
```

Comment:
Some remarks concerning our [coding style](https://www.openssl.org/policies/codingstyle.html):

- Allthough single line ifs should not be placed in braces, there is an exception from the roule: If either block of an if-else has braces, then the other one should get braces, too.
- there needs to be an empty line between variable declarations and code. Also, I don't know whether you placed the comma (instead of a semicolon) in line 224 on purpose, but the use of a comma expression appears strange to me. Instead of the `&=` assignment, you just could `&` the flag in the funmction call

```C
    if (no_public) {
        EC_KEY_set_ec_flags(eckey, EC_PKEY_NO_PUBKEY);
    } else {
        unsigned int enc_flags = EC_KEY_get_enc_flags(eckey);

        EC_KEY_set_enc_flags(eckey, enc_flags & ~EC_PKEY_NO_PUBKEY);
    }
```

Or why not simply like this?

```C
    if (no_public)
        EC_KEY_set_ec_flags(eckey, EC_PKEY_NO_PUBKEY);
    else
        EC_KEY_set_enc_flags(EC_KEY_get_enc_flags(eckey) & ~EC_PKEY_NO_PUBKEY);
```

Reply:
Ups, somehow the if-part got messed up. I will edit the last post.

Reply:
The comma is a mistake. The shorter is even better, thanks.

Reply:
By the way, this makes me wonder if the flag should be or'ed in the if-part, too?